core/platform/auth/firebase_auth_services.py

```
@@ -393,3 +366,105 @@ def associate_multi_auth_ids_with_user_ids(auth_id_user_id_pairs):
         auth_models.UserAuthDetailsModel.update_timestamps_multi(
             assoc_by_user_id_models)
         auth_models.UserAuthDetailsModel.put_multi(assoc_by_user_id_models)
+
+
+@contextlib.contextmanager
+def _suppress_firebase_errors():
+    """Returns a context manager to suppress Firebase Admin SDK exceptions.
+
+    Errors raised by the SDK are logged and then discarded.
+
+    Yields:
+        None. No relevent context expression.
+    """
+    try:
+        yield
+    except (ValueError, firebase_exceptions.FirebaseError):
+        logging.exception('Firebase Admin SDK raised an exception!')
```

Why are we suppressing these? I don't understand the context, and why this couldn't lead to suppressing legitimate errors...

Also should this just be part of firebase_admin_context? They seem to be called together all the time.

---

I've added a comment describing why we want to do this.

I'm unable to combine them because of limitations in `contextlib.contextmanager`. If we want to combine them, it'll have to be using a class-implementation (e.g. simple class with `__enter__` and `__exit__` methods).

Would that be an acceptable compromise?

---

Yup, totally fine with a more elaborate impl of contextmanager. You're thinking similar to CD in scripts/common.py, right?

---

As discussed, this isn't actually feasible due to the limitations of context managers.

If we do decide to go with this, the best I can probably do is create a wrapper context that establishes both of these at once.

---

Obsolete since we're no longer suppressing exceptions.